### PR TITLE
Ensure QNSPSA allows for batching

### PR DESCRIPTION
**Pull request by Cryoris**
### Summary

Batching as many circuits as possible into a single job is crucial for algorithm execution times on the real devices.
This PR ensures that using SPSA (QN-SPSA) with VQE allows for maximal batching, so unlike before only 1 (3) jobs are submitted per iteration and both calibration of the learning rate and the estimation of the standard deviation of the loss function use 1 job.

### Details and comments

To enable batching, the user must still set `max_evals_grouped` in the VQE, as before.

### CODE REVIEW

**Comment by woodsp-ibm** [comment]
We should have a reno I think - I guess this is a new feature.

**Comment by Cryoris** [comment]
Added 👍🏻

**Review by woodsp-ibm — DISMISSED** [review]

File: releasenotes/notes/ensure-qnspsa-batching-e48f7ec72412c071.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+features:
+  - |
+    Ensure that the :class:`~qiskit.algorithms.optimizers.SPSA` and 
+    :class:`~qiskit.algorithms.optimizers.QNSPSA` optimizers batch as many circuit
+    evaluations as possible for both the iterations and the initial calibrations.
```
Comment (mtreinish) [inline releasenotes/notes/ensure-qnspsa-batching-e48f7ec72412c071.yaml:6]: This could use some tweaking. Right now it reads like a bug fix release note that's missing the signposting details on the bug it's fixing. Instead what about something like:

```suggestion
    The :class:`~qiskit.algorithms.optimizers.SPSA` and
    :class:`~qiskit.algorithms.optimizers.QNSPSA` optimizer classes are now capable of
    batching as many circuit evaluations as possible for both the iterations and the initial
    calibrations. This can be leveraged by setting the ``max_evals_grouped`` kwarg on
    the constructor for :class:`~qiskit.algorithms.VQE` when using either
    :class:`~qiskit.algorithms.optimizers.SPSA` or
    :class:`~qiskit.algorithms.optimizers.QNSPSA` as the ``optimizer`` parameter. For example::
    
        from qiskit.circuit.library import TwoLocal
        from qiskit.algorithms import VQE
        from qiskit.algorithms.optimizers import QNSPSA
        from qiskit.test.mock import FakeMontreal
        
        backend = FakeMontreal()
        ansatz = TwoLocal(2, rotation_blocks=["ry", "rz"], entanglement_blocks="cz")
        qnspsa = QNSPSA(fidelity, maxiter=5)
        vqe = VQE(
            ansatz=ansatz,
            optimizer=qnspsa,
            max_evals_grouped=100,
            quantum_instance=backend,
        )
```

(feel free to update the example with something better, I was just kind of going off the unittest added)

Comment (Cryoris) [reply releasenotes/notes/ensure-qnspsa-batching-e48f7ec72412c071.yaml]: thanks for the addition 👍🏻

**Review by woodsp-ibm — APPROVED** [review]

**Review by mtreinish — APPROVED** [review]
